.github/workflows/main.yml

```diff
@@ -42,37 +56,76 @@ jobs:
         shell: bash
 
       - name: Build
-        run: dotnet run --project build/Build.csproj -- --target=Default
-
-      - name: Expose GitHub Runtime
-        uses: crazy-max/ghaction-github-runtime@v3
+        run: dotnet run --project build/Build.csproj -- --target=Default         
 
-      - name: Upload Artifacts
-        run: dotnet run --project build/Build.csproj -- --target=UploadArtifacts
-        env:
-          ACTIONS_RUNTIME_TOKEN: ${{ env.ACTIONS_RUNTIME_TOKEN }}
-          ACTIONS_RUNTIME_URL: "${{ env.ACTIONS_RUNTIME_URL }}"
   deploy:
-      name: Deploy
+      name: build-macos-latest and Optionally Publish Nugets
       needs: [ build ]
-      runs-on: ubuntu-latest
-      if: ${{ github.event_name == 'push' }}
+      runs-on: macos-latest
+
+      # if: ${{ github.event_name == 'push' }}
       permissions:
           packages: write
           contents: write
+
       steps:
-          - name: Clone Repository
-            uses: actions/checkout@v4
+        - name: Clone repository
+          uses: actions/checkout@v4
+          with:
+            submodules: recursive
+
+        - name: Setup .NET Core SDK 8
+          uses: actions/setup-dotnet@v4
+          with:
+            dotnet-version: '8.0.x'
+
+        - name: Setup Java
+          uses: actions/setup-java@v4
+          with:
+            distribution: 'microsoft'
+            java-version: '11'
+
+        - name: Setup NuGet
+          uses: nuget/setup-nuget@v2
+
+        - name: Disable annotations
+          run: echo "::remove-matcher owner=csc::"
+
+        - name: Install required workloads
+          run: |
+            dotnet workload install android ios macos
+          shell: bash
 
-          - name: Setup .NET ${{ matrix.dotnet-version }}
-            uses: actions/setup-dotnet@v4
-            with:
-              dotnet-version: '8.0.x'
+        - if: ${{ github.event_name == 'workflow_dispatch'}}
+          name: Check if tag contains '.Pre.'
+          id: check_preview
+          run: |
+            if [[ "${GITHUB_REF}" == *".Pre."* ]]; then
+              echo "is_preview=true" >> $GITHUB_OUTPUT
+            else
+              echo "is_preview=false" >> $GITHUB_OUTPUT
+            fi
+        
+        - if: ${{ github.ref != 'refs/heads/main' }}
+          name: Preview Build
+          run: dotnet run --project build/Build.csproj -- --target=Default --nuget-target=preview
 
-          - name: Expose GitHub Runtime
-            uses: crazy-max/ghaction-github-runtime@v3
+        - if: ${{ github.ref == 'refs/heads/main' }}
+          name: Release Build
+          run: dotnet run --project build/Build.csproj -- --target=Default
```

Comment:
The deploy action should do no building, its there to download artifacts from the build dashboard and upload them where its needed.